presto-spi/src/test/java/io/prestosql/spi/block/BaseBlockEncodingTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.block;
+
+import io.airlift.slice.DynamicSliceOutput;
+import io.prestosql.spi.type.Type;
+
+import static io.prestosql.spi.block.BlockTestUtils.assertBlockEquals;
+
+public abstract class BaseBlockEncodingTest<T>
+{
+    private final BlockEncodingSerde blockEncodingSerde = new TestingBlockEncodingSerde();
+
+    protected abstract Type getType();
+
+    protected abstract void write(BlockBuilder blockBuilder, T value);
+
+    protected final void roundTrip(Object... values)
```

Comment:
make this array of `T`

Reply:
Tried it, but I cannot make `new T[]` that way (testRandomData)
And then thought about List instead of array. But both `ImmutableList.of` and `List.of` does not accept nulls.
So this solutions seems the least bad